parser/src/ast.rs

```diff
@@ -0,0 +1,103 @@
+#[derive(Debug, PartialEq, Eq, Clone, Hash)]
+pub struct Located<T> {
+    pub l: usize,
```

Comment:
`l` indicates the position of the beginning of the word and `r` indicates the position of the end in the source text.
Your idea is good, but at the parsing stage it seemed to me that this approach is more intuitive, because the LALRPOP crate gives the location of the lexeme in this format